code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm

```diff
@@ -0,0 +1,176 @@
+// ***************************************
+// *********** Drain blood
+// ***************************************
+/datum/action/xeno_action/activable/drain
+	name = "Drain"
+	action_icon_state = "drain"
+	mechanics_text = "Hold a marine and drain some of their blood if successful."
+	ability_name = "drain"
+	cooldown_timer = 15 SECONDS
+	plasma_cost = 0
+
+/datum/action/xeno_action/activable/drain/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return
+	if(!iscarbon(A))
+		return FALSE
+	if(!A.can_sting())
+		to_chat(owner, span_xenowarning("This won't do!"))
+		return FALSE
+	var/mob/living/carbon/xenomorph/X = owner
+	if(X.plasma_stored >= X.xeno_caste.plasma_max)
+		to_chat(X, span_xenowarning("No need, we feel sated for now..."))
+		return FALSE
+	if(get_dist(X, A) > 1)
+		to_chat(X, span_xenowarning("It is outside of our reach! We need to be closer!"))
+		return FALSE
+	return TRUE
+
+/datum/action/xeno_action/activable/drain/use_ability(mob/living/carbon/A)
+	var/mob/living/carbon/xenomorph/X = owner
+	X.do_attack_animation(A, ATTACK_EFFECT_GRAB)
+	X.emote("roar")
+	A.drop_all_held_items()
+	X.visible_message(A, span_danger("The [X] grabs [A]!"))
+	A.SetImmobilized(2 SECONDS)
+	if(!do_after(X, 2 SECONDS, TRUE, A, BUSY_ICON_HOSTILE, ignore_turf_checks = FALSE))
+		to_chat(X, span_xenowarning("Our meal is interrupted!"))
+		succeed_activate()
+		add_cooldown()
+		return
+	X.do_attack_animation(A, ATTACK_EFFECT_REDSTAB)
+	A.emote("scream")
+	A.drop_all_held_items()
+	X.visible_message(A, span_danger("The [X] stabs its tail into [A]!"))
+	A.apply_damage(damage = 20, damagetype = BRUTE, def_zone = BODY_ZONE_HEAD, blocked = 0, sharp = TRUE, edge = FALSE, updating_health = TRUE)
+	playsound(A, "alien_claw_flesh", 25, TRUE)
+	X.gain_plasma(X.xeno_caste.drain_plasma_gain)
+	succeed_activate()
+	add_cooldown()
+
+// ***************************************
+// *********** Rejuvenate/Transfusion
+// ***************************************
+/datum/action/xeno_action/activable/rejuvenate
+	name = "Rejuvenate/Transfusion"
+	action_icon_state = "Rejuvenate"
+	mechanics_text = "When used on self, drains blood and restores health over time. When used on another xenomorph, costs blood and restores some of their health. When used on a dead human, you heal gradually."
```

Comment:
all these missiong action icons rn see actions.dmi we have a lot of unused ones you might find something

Reply:
fixed